pallets/parachain-staking/src/types.rs

```diff
@@ -1362,6 +1383,49 @@ impl<
 		Err(Error::<T>::DelegationDNE.into())
 	}
 
+	/// Updates the bond locks for this delegator.
+	///
+	/// This will take the current self.total and ensure that a lock of the same amount is applied
+	/// and will also ensure that the account has at least `additional_reserve` in free balance.
+	///
+	/// `additional_required_balance` should reflect the change to the amount that should be locked if
+	/// positive, 0 otherwise (e.g. `min(0, change_in_total_bond)`). This is necessary because it is
+	/// not possible to query the amount that is locked for a given lock id.
+	pub fn adjust_bond_lock<T: Config>(
+		&mut self,
+		additional_required_balance: Option<Balance>,
```

Comment:
I gave it additional thought and I believe an enum defines this input better:

```rust
pub enum BondAdjust<Balance> {
	Increase(Balance),
	Decrease,
}
```

which clearly conveys that an increase would be checked, but a decrease would not be.

I also tried to theorycraft a coherent process to update total in-sync with locked value here using closures and amde `total` a private field https://github.com/PureStake/moonbeam/compare/notlesh-staking-locks...nish-notlesh-staking-locks. 

Would be great to discuss on it.

Reply:
I like this idea, I'll give it some consideration soon. I also played with merging the disparate functions of updating the `self.total`, the locks, and the storage item, but haven't gotten anywhere with it yet.